### PR TITLE
Migrate 'Global styles sidebar' test to Playwright

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.

PR migrates 'Global styles sidebar' tests to Playwright.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/site-editor/global-styles-sidebar.spec.js
```

### CODE REVIEW

**Review by ntsekouras — APPROVED** [review]
Looks good, thank you!

**Comment by Mamaduka** [comment]
Thanks for the review, Nik!

File: test/e2e/specs/site-editor/global-styles-sidebar.spec.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			.fill( 'heading' );
+
+		// Matches both Heading and Table of Contents blocks.
+		// The latter contains "heading" in its description.
+		await expect(
+			page.getByRole( 'button', { name: 'Heading block styles' } )
+		).toBeVisible();
+		await expect(
+			page.getByRole( 'button', {
+				name: 'Table of Contents block styles',
```
Comment (ellatrix) [inline test/e2e/specs/site-editor/global-styles-sidebar.spec.js:45]: Can we use another block here? The Table of Contents block is not in core, so this test will fail when it's run without GB active.

Comment (ellatrix) [reply test/e2e/specs/site-editor/global-styles-sidebar.spec.js:45]: I guess it already checks for Heading, so maybe we can just remove this check

Comment (ellatrix) [reply test/e2e/specs/site-editor/global-styles-sidebar.spec.js:45]: Or use another keyword if we need multiple block results

Comment (Mamaduka) [reply test/e2e/specs/site-editor/global-styles-sidebar.spec.js:45]: Here's the follow-up - #73685.

**Pull request merged**